src/Aspire.Hosting.Sdk/SDK/Sdk.targets

```diff
@@ -20,7 +18,21 @@
     <IsPackable Condition="'$(IsPackable)' == ''">false</IsPackable>
   </PropertyGroup>
 
-  <!-- *** END ***  -->
+  <!--
+  Default AppHost ProjectReference items to not be referenced in the AppHost, unless it is AspireHostLibrary=true.
+  This defaulting needs to happen in the SDK targets so this metadata affects NuGet Restore.
+  -->
+  <ItemGroup Condition="'$(IsAspireHost)' == 'true'">
+    
+    <ProjectReference Update="@(ProjectReference)">
+      <_IsServiceApp Condition="'%(_IsServiceApp)' == '' and '%(AspireHostLibrary)' != 'true'">true</_IsServiceApp>
+
+      <ReferenceOutputAssembly Condition="'%(ReferenceOutputAssembly)' == '' and '%(_IsServiceApp)' == 'true'">false</ReferenceOutputAssembly>
+      <SkipGetTargetFrameworkProperties Condition="'%(SkipGetTargetFrameworkProperties)' == '' and '%(_IsServiceApp)' == 'true'">true</SkipGetTargetFrameworkProperties>
```

Comment:
I'm assuming this works transitively, right? Like if AppHost references a project which then references another project?

Reply:
Correct - that's the intention. This is why I needed to put this logic in the Sdk.targets file, so NuGet Restore doesn't transitively add references of the referenced projects.